test/ruby/test_zjit.rb

```diff
@@ -1040,15 +1040,47 @@ def test(x)
     }, insns: [:opt_newarray_send], call_threshold: 1
   end
 
+  def test_opt_newarray_send_include_p_redefinition
+    assert_compiles '[true, false]', %q{
+      class Array
+        alias_method :old_include?, :include?
+        def include?(x)
+          old_include?(x)
+        end
+      end
+
+      def test(x)
+        [:y, 1, Object.new].include?(x)
+      end
+      [test(1), test("n")]
+    }, insns: [:opt_newarray_send], call_threshold: 1
+  end
+
   def test_opt_duparray_send_include_p
     assert_compiles '[true, false]', %q{
+      class Array
+        alias_method :old_include?, :include?
```

Comment:
I think you accidentally swapped the bodies/names of this test and the one below it